src/functions/href.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import defineFunction, {ordargument} from "../defineFunction";
+import buildCommon from "../buildCommon";
+import mathMLTree from "../mathMLTree";
+
+import * as html from "../buildHTML";
+import * as mml from "../buildMathML";
+
+defineFunction({
+    type: "href",
+    names: ["\\href"],
+    props: {
+        numArgs: 2,
+        argTypes: ["url", "original"],
+    },
+    handler: (context, args) => {
+        const body = args[1];
+        const href  = args[0].value;
+        return {
+            type: "href",
+            href: href,
+            body: ordargument(body),
+        };
+    },
+    htmlBuilder: (group, options) => {
+        const elements = html.buildExpression(
+            group.value.body,
+            options,
+            false
+        );
+
+        const href = group.value.href;
+
+        /**
+         * Determining class for anchors.
+         * 1. if it has the only element, use its class;
+         * 2. if it has more than two elements, and the classes
+         *    of its first and last elements coincide, then use it;
+         * 3. otherwise, we will inject an empty <span>s at both ends,
+         *    with the same classes of both ends of elements.
+         */
+
+        let classes = []; // null if the type of both ends differs.
+        let first; // mathtype of the first child
+        let last; // mathtype of the last child
+        // Invariants: both first and last must be non-null if classes is null.
+        if (elements.length === 1) { // Case 1
+            classes = elements[0].classes;
+        } else if (elements.length >= 2) {
+            first = html.getTypeOfDomTree(elements[0]) || 'mord';
+            last = html.getTypeOfDomTree(elements[elements.length - 1]) || 'mord';
+            if (first === last) { // Case 2 : type of both ends coincides
+                classes = [first];
+            } else { // Case 3: both ends have different types.
+                classes = null;
+            }
+        } else { // No elements at all, just ignore.
+            classes = [];
+        }
+        if (!classes) {
+            const anc = buildCommon.makeAnchor(href, [], elements, options);
+            return new buildCommon.makeFragment([
```

Comment:
This works, but I'd probably move this up to where you set `classes = null` so that we can remove an extra `if` statement.

Reply:
Oh, yes. It makes program much simpler. Thanks.